frontend/src/metabase/admin/performance/components/StrategyForm.tsx

```diff
@@ -0,0 +1,293 @@
+import { useFormikContext } from "formik";
+import type { ReactNode } from "react";
+import { useEffect, useState } from "react";
+import { t } from "ttag";
+import _ from "underscore";
+
+import type { FormTextInputProps } from "metabase/forms";
+import {
+  Form,
+  FormProvider,
+  FormRadioGroup,
+  FormSubmitButton,
+  FormTextInput,
+  useFormContext,
+} from "metabase/forms";
+import { color } from "metabase/lib/colors";
+import {
+  Button,
+  Group,
+  Icon,
+  Loader,
+  Radio,
+  Stack,
+  Text,
+  Title,
+  Tooltip,
+} from "metabase/ui";
+
+import { rootId } from "../constants";
+import { useRecentlyTrue } from "../hooks/useRecentlyTrue";
+import type { Strategy, StrategyType } from "../types";
+import { DurationUnit, Strategies } from "../types";
+import { strategyValidationSchema } from "../validation";
+
+export const StrategyForm = ({
+  targetId,
+  setIsDirty,
+  saveStrategy,
+  savedStrategy,
+}: {
+  targetId: number | null;
+  setIsDirty: (isDirty: boolean) => void;
+  saveStrategy: (values: Strategy) => Promise<void>;
+  savedStrategy?: Strategy;
+}) => {
+  const defaultStrategy: Strategy = {
+    type: targetId === rootId ? "nocache" : "inherit",
+  };
+  return (
+    <FormProvider<Strategy>
+      key={targetId}
+      initialValues={savedStrategy ?? defaultStrategy}
+      validationSchema={strategyValidationSchema}
+      onSubmit={saveStrategy}
+      enableReinitialize
+    >
+      <StrategyFormBody targetId={targetId} setIsDirty={setIsDirty} />
+    </FormProvider>
+  );
+};
+
+const StrategyFormBody = ({
+  targetId,
+  setIsDirty,
+}: {
+  targetId: number | null;
+  setIsDirty: (isDirty: boolean) => void;
+}) => {
+  const { dirty, values, setFieldValue } = useFormikContext<Strategy>();
+  const { setStatus } = useFormContext();
+  const [wasDirty, setWasDirty] = useState(false);
+
+  const selectedStrategyType = values.type;
+
+  useEffect(() => {
+    setIsDirty(dirty);
+  }, [dirty, setIsDirty]);
+
+  useEffect(() => {
+    // When form becomes dirty, reset form status to idle
+    setWasDirty(dirty);
+    if (dirty && !wasDirty) {
+      setStatus("idle");
+    }
+  }, [dirty, wasDirty, setIsDirty, setStatus]);
+
+  useEffect(() => {
+    if (selectedStrategyType === "duration") {
+      setFieldValue("unit", DurationUnit.Hours);
+    }
+  }, [selectedStrategyType, values, setFieldValue]);
+
+  return (
+    <Form
+      h="100%"
+      style={{
+        display: "flex",
+        flexDirection: "column",
+        justifyContent: "space-between",
+      }}
+    >
+      <Stack p="lg" spacing="xl" maw="27.5rem">
+        <StrategySelector targetId={targetId} />
+        {selectedStrategyType === "ttl" && (
```

Comment:
This is fine for now, but when we have more than a few different strategies and some of them are in enterprise and some are not, I wonder if it might make sense to just have totally different form components for each strategy. That might also obviate the necessity for some of the form resets you're doing elsewhere: you don't need to swap around which fields are visible in the form if you just have different forms.